compat/src/index.d.ts

```diff
@@ -284,14 +284,26 @@ declare namespace React {
 		ComponentProps<T>
 	>;
 
-	export type ComponentPropsWithRef<
-		C extends ComponentType<any> | keyof JSXInternal.IntrinsicElements
-	> = C extends new (
+	export type ComponentPropsWithRef<C extends ElementType> = C extends new (
```

Comment:
We didn't originally have `ElementType` when I added `ComponentPropsWithRef` but someone added it a few months ago so it's worth switching this over, helps ensure things are kept in-sync.

For comparison: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b5dc32740d9b45d11cff9b025896dd333c795b39/types/react/index.d.ts#L1687-L1689